system/Helpers/form_helper.php

```diff
@@ -739,7 +739,7 @@ function validation_show_error(string $field, string $template = 'single'): stri
         $config = config(Validation::class);
         $view   = service('renderer');
 
-        $errors = array_filter(validation_errors(), static fn ($key) => preg_match(
+        $errors = array_filter(validation_errors(), static fn ($key): false|int => preg_match(
```

Comment:
array_filter accepts a bool returning function. so i think this should be bool and the preg_match is matched against !== 1 again?

Reply:
should be compare to 1 directly since it is not negation, updated https://github.com/codeigniter4/CodeIgniter4/pull/9343/commits/f29210c0f0ef4c4096d4ac937f1e422ef769673f